packages/base/all/vendor-config-onl/src/python/onl/install/SystemInstall.py

```diff
@@ -0,0 +1,307 @@
+"""App.py
+
+Application code for swl-install.
+"""
+
+import logging
+import os, sys
+import json
+import tempfile
+import zipfile
+import shutil
+import argparse
+import fnmatch
+import subprocess
+
+from onl.install.InstallUtils import InitrdContext, MountContext
+from onl.install.InstallUtils import BlkidParser, ProcMountsParser
+from onl.install.ConfUtils import MachineConf, InstallerConf
+from onl.install.ShellApp import Onie, Upgrader
+import onl.install.App
+from onl.sysconfig import sysconfig
+from onl.platform.current import OnlPlatform
+
+from onl.install.InstallUtils import SubprocessMixin
+
+class UpgradeHelper(Upgrader):
+
+    def __init__(self, callback=None, log=None):
+        super(UpgradeHelper, self).__init__(log=log)
+        self.callback = callback
+
+    def _runInitrdShell(self, p):
+        if self.callback is not None:
+            self.callback(self, p)
+
+class OnlBootContext(object):
```

Comment:
You should use the existing OnlMountReadWriteContext("ONL-BOOT") from onl.mounts